test/utils/create_resources.go

```diff
@@ -54,19 +53,6 @@ func RetryWithExponentialBackOff(fn wait.ConditionFunc) error {
 	return wait.ExponentialBackoff(backoff, fn)
 }
 
-func IsRetryableAPIError(err error) bool {
-	// These errors may indicate a transient error that we can retry in tests.
-	if apierrors.IsInternalError(err) || apierrors.IsTimeout(err) || apierrors.IsServerTimeout(err) ||
-		apierrors.IsTooManyRequests(err) || utilnet.IsProbableEOF(err) || utilnet.IsConnectionReset(err) {
```

Comment:
Hmm - actually I started wondering about that.

The first 4 things actually really mean that there is something wrong happening on Kubernetes side.
But what about the last two - those are purely http-level stuff. And they will be happening even if Kubernetes would be perfect, right?

So do we really want to get rid of that kind-of deflaking mechanism?

Reply:
if they mask problems with the cluster and keep us from catching issues because some of our test environments are unstable, I'd rather know about that and fix the test environments

Reply:
I'm not sure these are connected to test environments being unstable. Connections are breaking sporadically, right? The http-level ones can not be because of Kubernetes is doing something wrong, but because underlying infrastructure isn't perfect. I'm not sure we want to penalize us for it (in other words, I'm not sure if we will get something actionable on our side from those errors all the time - this is only about the utilnet errors, I support removing the first 4 k8s-related).

Reply:
If the environments don't have stable tcp behavior, then that can manifest in many places that don't hold blocks like these and probably never will, including
1. any proxy behavior in the server
2. retry behavior for various kinds of webhooks
3. kube-apiserver to kubelet
4. kubelet to kube-apiserver
5. kube-controller-manager to kube-apiserver
6. kube-scheduler to kube-apiserver
7. kubectl (possibly from our e2es, but also in general) to kube-apiserver

This set of retries here can move those TCP failures from being easy to identify and locate in our e2e tests to side-effect behaviors in other locations.  I found this because there appear to be cases in some openshift build-farms where we experience network level weirdness.  I would rather fail early and find environmental problems.

For some classes of these failures, I'm in favor of placing retry logic in client-go, but for the ones that are not going to go into client-go, I would rather fail early and consistently.

Reply:
Our components has to accept that errors may happen. If our controllers get an error, they will reque the processing of an object and will process it again later (or at least they should).
This is not the case in our tests - they generally fail, sometime in unexpected ways. Maybe that's fine (and we will fix those unexpected ways along the way), but what I'm saying that I think those are two different situations imho.

Reply:
> Our components has to accept that errors may happen. If our controllers get an error, they will reque the processing of an object and will process it again later (or at least they should).
> This is not the case in our tests - they generally fail, sometime in unexpected ways. Maybe that's fine (and we will fix those unexpected ways along the way), but what I'm saying that I think those are two different situations imho.

If we want to have uniform retry behavior in tests (and I'm not fully convinced that we do), then it seems more reasonable to build those retries into a configuration retry method on clients.  That would allow uniform retries instead of one-off uses of this method.

I think that the problems we chase in tests are rarely due to network infrastructure hiccups and that given how short running these tests are, such a hiccup would point to infrastructure problems we want to be aware of because they will impact our general topology.

Reply:
> ... it seems more reasonable to build those retries into a configuration retry method on clients. That would allow uniform retries instead of one-off uses of this method.

Agree

> I think that the problems we chase in tests are rarely due to network infrastructure hiccups and that given how short running these tests are, such a hiccup would point to infrastructure problems we want to be aware of because they will impact our general topology.

That I'm not sure I fully agree. But I also don't feel strongly about that, so I'm not going to block this PR.